sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py

```diff
@@ -200,11 +208,13 @@ def _wrap_eventdata(
                     raise ValueError(
                         "The partition_key does not match the one of the EventDataBatch"
                     )
+                for message in event_data.message._body_gen:  # pylint: disable=protected-access
+                    add_link_to_send(message, span)
                 wrapper_event_data = event_data  # type:ignore
             else:
                 if partition_key:
                     event_data = _set_partition_key(event_data, partition_key)
-                event_data = _set_trace_message(event_data)
+                event_data = _set_trace_message(event_data, span)
```

Comment:
do we need `add_link_to_send` to all the events in event_data here? event_data is type of `Iterable[EventData]`,